store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/handler/spark/GetDataFrameOfElementsHandler.java

```diff
@@ -0,0 +1,81 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.parquetstore.operation.handler.spark;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.StructType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.gaffer.data.elementdefinition.view.View;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.parquetstore.ParquetStore;
+import uk.gov.gchq.gaffer.parquetstore.utils.SchemaUtils;
+import uk.gov.gchq.gaffer.spark.SparkContextUtil;
+import uk.gov.gchq.gaffer.spark.operation.dataframe.GetDataFrameOfElements;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
+
+import java.io.Serializable;
+
+/**
+ * An {@link uk.gov.gchq.gaffer.store.operation.handler.OperationHandler} for the {@link GetDataFrameOfElements}
+ * operation on the {@link ParquetStore}.
+ */
+public class GetDataFrameOfElementsHandler implements OutputOperationHandler<GetDataFrameOfElements, Dataset<Row>>, Serializable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetDataFrameOfElementsHandler.class);
+    private static final long serialVersionUID = 6355341028414862100L;
+
+    @Override
+    public Dataset<Row> doOperation(final GetDataFrameOfElements operation,
+                                    final Context context,
+                                    final Store store) throws OperationException {
+        final SparkSession spark = SparkContextUtil.getSparkSession(context, store.getProperties());
+        return doOperation(operation, (ParquetStore) store, spark);
+    }
+
+    private Dataset<Row> doOperation(final GetDataFrameOfElements operation,
+                                     final ParquetStore store,
+                                     final SparkSession spark) throws OperationException {
+        if (!operation.getView()
+                .equals(new View.Builder()
+                        .entities(store.getSchema().getEntityGroups())
+                        .edges(store.getSchema().getEdgeGroups()).build())) {
+            throw new OperationException("This operation does not currently support views");
+        }
+        LOGGER.debug("Creating a Dataset<Row> from path {} with option mergeSchema=true", store.getGraphPath());
+        final StructType schema = new SchemaUtils(store.getSchema()).getMergedSparkSchema(store.getSchema().getGroups());
+        final Dataset<Row> dataframe = spark
+                .read()
+                .schema(schema)
+                .parquet(store.getGraphPath());
+        return dataframe;
+    }
+
+//    private boolean isVisible(final Row e, final String visibility, final Authorisations auths) throws VisibilityParseException {
```

Comment:
Can you remove this

Reply:
Moved to the issue to reintroduce visibility filtering so that the code is not lost.

Reply:
This should still not be added to the develop branch. If you want to save the code, it should be put in a separate local branch and then removed from this one.